aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -354,16 +352,26 @@ Tensor flip_mps(const Tensor& self, IntArrayRef dims) {
           MPSGraphTensor* inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, self);
           MPSGraphTensor* indexTensor = mpsGraphRankedPlaceHolder(mpsGraph, index);
           MPSGraphTensor* sourceTensor = mpsGraphRankedPlaceHolder(mpsGraph, source);
-          MPSGraphTensor* alphaTensor = mpsGraphScalarPlaceHolder(mpsGraph, alpha_f);
-          MPSGraphTensor* alphaSourceSlice = [mpsGraph multiplicationWithPrimaryTensor:sourceTensor
+          MPSGraphTensor* alphaTensor = mpsGraphScalarPlaceHolder(mpsGraph, getMPSScalarType(kFloat));
```

Comment:
Hi @qqaatw, thanks for the PR. We should stick within floating types for promoting to FP32. For int types you can use i32 . And for i64 we should raise the error. (Although this is fixed in Ventura) 
src_type = isFloatingType(source.scalar_type()) ? MPSDataTypeFloat32 : MPSDataTypeInt32;

Reply:
Done.